allennlp/training/model.py

```diff
@@ -0,0 +1,89 @@
+from typing import Any, List, Dict, Union
+import torch
+
+
+class Model(torch.nn.Module):
+    """
+    This abstract class represents a model to be trained. Rather than relying on the
+    raw torch Module, we provide a slightly different API.
+    """
+
+    def forward(self,
```

Comment:
Yeah, I like the last idea - just always return a value for loss, and if there are no targets, that loss is `None`.